lib/src/api/engine/remote/http/mod.rs

```diff
@@ -192,21 +188,23 @@ async fn query(request: RequestBuilder) -> Result<QueryResponse> {
 
 async fn take(one: bool, request: RequestBuilder) -> Result<Value> {
 	if let Some(result) = query(request).await?.0.remove(&0) {
-		let mut vec = result?;
+		let result = result?;
 		match one {
-			true => match vec.pop() {
-				Some(Value::Array(Array(mut vec))) => {
+			true => match result {
+				Value::Array(Array(mut vec)) => {
 					if let [value] = &mut vec[..] {
 						return Ok(mem::take(value));
 					}
 				}
-				Some(Value::None | Value::Null) | None => {}
-				Some(value) => {
+				Value::None | Value::Null => {}
+				value => {
 					return Ok(value);
 				}
 			},
 			false => {
-				return Ok(Value::Array(Array(vec)));
+				// TODO should this return a vec?
```

Comment:
Same here.

Reply:
Tests failed because I made it a vec at first last night, but I forgot to remove my comment. Resolving now